src/display/svg.js

```diff
@@ -423,1061 +434,1045 @@ SVGGraphics = (function SVGGraphicsClosure() {
+    } else {
+      clipElement.setAttributeNS(null, 'clip-rule', 'nonzero');
+    }
+    this.pendingClip = null;
+    clipPath.appendChild(clipElement);
+    this.defs.appendChild(clipPath);
+
+    if (current.activeClipUrl) {
+      // The previous clipping group content can go out of order -- resetting
+      // cached clipGroups.
+      current.clipGroup = null;
+      this.extraStack.forEach(function(prev) {
+        prev.clipGroup = null;
+      });
+      // Intersect with the previous clipping path.
+      clipPath.setAttributeNS(null, 'clip-path', current.activeClipUrl);
+    }
+    current.activeClipUrl = `url(#${clipId})`;
+
+    this.tgrp = null;
+  }
+
+  clip(type) {
+    this.pendingClip = type;
+  }
+
+  closePath() {
+    const current = this.current;
+    if (current.path) {
+      const d = `${current.path.getAttributeNS(null, 'd')}Z`;
       current.path.setAttributeNS(null, 'd', d);
-      current.path.setAttributeNS(null, 'fill', 'none');
+    }
+  }
 
-      // Saving a reference in current.element so that it can be addressed
-      // in 'fill' and 'stroke'
-      current.element = current.path;
-      current.setCurrentPoint(x, y);
-    },
+  setLeading(leading) {
+    this.current.leading = -leading;
+  }
 
-    endPath: function SVGGraphics_endPath() {
-      // Painting operators end a path.
-      this.current.path = null;
+  setTextRise(textRise) {
+    this.current.textRise = textRise;
+  }
 
-      if (!this.pendingClip) {
-        return;
-      }
-      var current = this.current;
-      // Add current path to clipping path
-      var clipId = 'clippath' + clipCount;
-      clipCount++;
-      var clipPath = this.svgFactory.createElement('svg:clipPath');
-      clipPath.setAttributeNS(null, 'id', clipId);
-      clipPath.setAttributeNS(null, 'transform', pm(this.transformMatrix));
-      // A deep clone is needed when text is used as a clipping path.
-      const clipElement = current.element.cloneNode(true);
-      if (this.pendingClip === 'evenodd') {
-        clipElement.setAttributeNS(null, 'clip-rule', 'evenodd');
-      } else {
-        clipElement.setAttributeNS(null, 'clip-rule', 'nonzero');
-      }
-      this.pendingClip = null;
-      clipPath.appendChild(clipElement);
-      this.defs.appendChild(clipPath);
-
-      if (current.activeClipUrl) {
-        // The previous clipping group content can go out of order -- resetting
-        // cached clipGroups.
-        current.clipGroup = null;
-        this.extraStack.forEach(function (prev) {
-          prev.clipGroup = null;
-        });
-        // Intersect with the previous clipping path.
-        clipPath.setAttributeNS(null, 'clip-path', current.activeClipUrl);
+  setTextRenderingMode(textRenderingMode) {
+    this.current.textRenderingMode = textRenderingMode;
+  }
+
+  setHScale(scale) {
+    this.current.textHScale = scale / 100;
+  }
+
+  setGState(states) {
+    for (const state of states) {
```

Comment:
Unless you actually need the `state` variable below, you could probably just do `for (const [key, value] of states) {` instead.

Reply:
Fixed in the new commit.